src/assign_gvi_to_points.py

```diff
@@ -0,0 +1,107 @@
+"""Assign Green View score to point features"""
+
+import os
+import cv2
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+from skimage.filters import threshold_otsu
+import typer
+
+app = typer.Typer()
+
+
+def get_gvi_score(image_path):
+    """
+    Calculate the Green View Index (GVI) for a given image file.
+
+    Args:
+        image_path (str): Path to the image file.
+
+    Returns:
+        float: The Green View Index (GVI) score for the given image.
+    """
+    # Load the image
+    original_image = cv2.imread(image_path)
+
+    # Convert to RGB color space
+    rgb_image = cv2.cvtColor(original_image, cv2.COLOR_BGR2RGB)
+
+    # Calculate ExG (Excess Green)
+    r, g, b = cv2.split(rgb_image.astype(np.float32) / 255)
+    exg = 2 * g - r - b
+
+    # Apply Otsu's thresholding on ExG
+    threshold = threshold_otsu(exg)
+    green_pixels = (exg > threshold).sum()
+    total_pixels = original_image.shape[0] * original_image.shape[1]
+
+    # Calculate the Green View Index (GVI)
+    gvi_score = (green_pixels / total_pixels) * 100
+
+    return gvi_score
+
+
+@app.command()
+def main(image_directory, interim_data, output_file):
+    """
```

Comment:
nit: You can use `Annotated` to build out the help message. See https://github.com/AmericanRedCross/street-view-green-view/blob/main/src/download_images.py#L132.

Reply:
Thanks - added help annotations